docs/web3-eth-admin.rst

```diff
@@ -0,0 +1,423 @@
+.. _eth-admin:
+
+.. include:: include_announcement.rst
+
+==============
+web3.eth.admin
+==============
+
+
+The ``web3-eth-admin`` package allows you to interact with the Ethereum node's admin management.
+
+
+.. code-block:: javascript
+
+    import Web3 from 'web3';
+    import {Admin} from 'web3-eth-admin';
+
+    // "Web3.givenProvider" will be set if in an Ethereum supported browser.
+    const admin = new Admin(Web3.givenProvider || 'ws://some.local-or-remote.node:8546', null, options);
+
+
+    // or using the web3 umbrella package
+    const web3 = new Web3(Web3.givenProvider || 'ws://some.local-or-remote.node:8546', null, options);
+
+    // -> web3.eth.admin
+
+
+------------------------------------------------------------------------------
+
+
+.. include:: include_package-core.rst
+
+
+
+------------------------------------------------------------------------------
+
+.. _admin-addpeer:
+
+addPeer
+=========
+
+.. code-block:: javascript
+
+    web3.eth.admin.addPeer(url, [callback])
+
+Add an admin peer on the node that Web3 is connected to with its provider.
+The RPC method used is ``admin_addPeer``.
+
+----------
+Parameters
+----------
+
+1. ``url`` - ``String``:  The enode URL of the remote peer.
+1. ``Function`` - (optional) Optional callback, returns an error object as first parameter and the result as second.
+
+-------
+Returns
+-------
+
+``Promise<boolean>`` - True if peer added successfully.
+
+-------
+Example
+-------
+
+.. code-block:: javascript
+
+    web3.eth.admin.addPeer("enode://a979fb575495b8d6db44f750317d0f4622bf4c2aa3365d6af7c284339968eef29b69ad0dce72a4d8db5ebb4968de0e3bec910127f134779fbcb0cb6d3331163c@52.16.188.185:30303")
+    .then(console.log);
+    > true
+
+------------------------------------------------------------------------------
+
+
+getDataDirectory
+=====================
+
+.. code-block:: javascript
+
+    web3.eth.admin.getDataDirectory([, callback])
+
+Provides absolute path of the running node, which is used by the node to store all its databases.
+
+----------
+Parameters
+----------
+
+
+1. ``Function`` - (optional) Optional callback, returns an error object as first parameter and the result as second.
+
+
+-------
+Returns
+-------
+
+
+``Promise<string>`` - The path.
+
+
+-------
+Example
+-------
+
+
+.. code-block:: javascript
+
+    web3.eth.admin.getDataDirectory()
+    .then(console.log);
+    > "/home/ubuntu/.ethereum"
+
+
+------------------------------------------------------------------------------
+
+
+getNodeInfo
+=====================
+
+.. code-block:: javascript
+
+    web3.eth.personal.getNodeInfo([, callback])
+
+This property can be queried for all the information known about the running node at the networking granularity..
+
+----------
+Parameters
+----------
+
+
+1. ``Function`` - (optional) Optional callback, returns an error object as first parameter and the result as second.
+
+
+-------
+Returns
+-------
+
+
+``Promise<object>`` - The node information array.
```

Comment:
Please detail the object call more with the object and a description next to each property.